ui/src/message_popup/mobile_prototype/mobile_interface.jsx

```diff
@@ -0,0 +1,383 @@
+        key: 'question'
+      };
+      messages.push(question);
+      if(nextProps.shouldShowStudentCard && _.has(nextProps.question, 'studentId')){
+        const studentObject = (withStudents([nextProps.question]))[0].student
+        var text = studentObject.name + "\n"
+                      + studentObject.grade + " " + studentObject.gender + ", " + studentObject.race;
+        if(_.has(studentObject, 'behavior') && studentObject.behavior !== null){ text+= "\n" + studentObject.behavior };
+        if(_.has(studentObject, 'ell') && studentObject.ell !== null){ text+= "\n" + studentObject.ell };
+        if(_.has(studentObject, 'learningDisabilities') && studentObject.learningDisabilities !== null){ text+= "\n" + studentObject.learningDisabilities };
+        if(_.has(studentObject, 'academicPerformance') && studentObject.academicPerformance !== null){ text+= "\n" + studentObject.academicPerformance };
+        if(_.has(studentObject, 'interests') && studentObject.interests !== null){ text+= "\n" + studentObject.interests };
+        if(_.has(studentObject, 'familyBackground') && studentObject.familyBackground !== null){ text+= "\n" + studentObject.familyBackground };
+        if(_.has(studentObject, 'ses') && studentObject.ses !== null){ text+= "\n" + studentObject.ses };
+        const student = {
+          type: 'student',
+          text: text,
+          key: 'studentInfo'
+        }
+        messages.push(student);
+      }
+    }
+    return messages;
+  },
+  
+  componentDidMount() {
+    setInterval(this.updateTimer, ONE_SECOND);
+  },
+
+  updateTimer() {
+    if(!this.isReadyToMoveOn()){
+      this.setState({ elapsedMs: this.state.elapsedMs + ONE_SECOND });
+    }
+  },
+  
+  componentWillMount(){
+    this.messageId = parseInt(this.props.question.id)*10 + 1;
+  },
+  
+  componentWillReceiveProps(nextProps){
+    var messages = this.getInitialMessages(nextProps);
+    const goodExamples = _.map(nextProps.question.examples, (example) => {
+      return {type: 'Good', text: example};
+    });
+    const badExamples = _.map(nextProps.question.nonExamples, (example) => {
+      return {type: 'Bad', text: example};
+    });
+    const allExamples = _.shuffle(_.clone(goodExamples).concat(badExamples)); 
+    //this.addMessages(messages);
+    const elapsedMs = 0;
+    this.setState({ messages, allExamples, elapsedMs: 0 });
+  },
+  
+  componentDidUpdate(prevProps, prevState){
+    if(prevState.elapsedMs === 90000){
+      this.addMessages([{type: 'info', text: 'Remember, these are supposed to be quick responses.'}]);
+    }
+    if(prevState.initialResponse === undefined && this.state.initialResponse !== undefined){
+      this.logResponse();
+    }else if(prevState.revisedResponse === undefined && this.state.revisedResponse === ''){
+      this.logRevisionDeclined();
+    }else if(prevState.revisedResponse === undefined && this.state.revisedResponse !== undefined){
+      this.logRevision(this.state.revisedResponse);
+    }
```

Comment:
Let's see if we can do this at the call site, where we handle the action that the user is taking, instead of having to compute what has changed here.